polkadot/xcm/src/v3/multiasset.rs

```diff
@@ -694,7 +694,9 @@ impl MultiAssets {
 		target: &MultiLocation,
 		context: InteriorMultiLocation,
 	) -> Result<(), ()> {
-		self.0.iter_mut().try_for_each(|i| i.reanchor(target, context))
+		self.0.iter_mut().try_for_each(|i| i.reanchor(target, context))?;
+		self.0.sort();
```

Comment:
I would also warn of this on its doc-comment, smth like:
```
	/// Mutate the location of the asset identifier if concrete, giving it the same location
	/// relative to a `target` context. The local context is provided as `context`.
	/// This will also re-sort the inner assets to preserve asset ids ordering guarantees.
	pub fn reanchor(
```

Reply:
Done.